afpy/models/NewsEntry.py

```diff
@@ -0,0 +1,94 @@
+
+from afpy.models import BaseModel
+from afpy.models.AdminUser import AdminUser
+
+
+class NewsEntry(BaseModel):
+    title = TextField(null=False, help_text="Title of the news entry", verbose_name="Title")
+    summary = TextField(null=True, help_text="Summary of the news entry", verbose_name="Summary")
+    content = TextField(null=False, help_text="Content of the news entry", verbose_name="Content")
+    dt_submitted = DateTimeField(
+        null=False,
+        default=datetime.now,
+        help_text="When was the news entry submitted",
+        verbose_name="Datetime Submitted",
+    )
+    dt_updated = DateTimeField(
+        null=False, default=datetime.now, help_text="When was the news entry updated", verbose_name="Datetime Updated"
+    )
+    dt_published = DateTimeField(
+        null=True, help_text="When was the news entry published", verbose_name="Datetime Published"
+    )
+    state = CharField(
+        null=False,
+        default="waiting",
+        choices=[("waiting", "waiting"), ("published", "published"), ("rejected", "rejected")],
+        help_text="Current state of the news entry",
+        verbose_name="State",
+    )
+    approved_by = ForeignKeyField(
+        AdminUser,
+        null=True,
+        default=None,
+        backref="adminuser",
+        help_text="Who approved the news entry",
+        verbose_name="Approved by",
+    )
+    author = CharField(null=False, default="Admin", help_text="Author of the news entry", verbose_name="Author")
+    author_email = CharField(null=False, help_text="Author email", verbose_name="Author Email")
+    image_url = TextField(null=True, help_text="Image for the news entry", verbose_name="Image URL")
+
+    @classmethod
+    def create(
+        cls,
+        title: str,
+        content: str,
+        author: str,
+        author_email: str,
+        image_url: Optional[str] = None,
+        summary: Optional[str] = None,
+        dt_submitted: Optional[datetime] = None,
+        dt_updated: Optional[datetime] = None,
+        dt_published: Optional[datetime] = None,
+        state: str = "waiting",
+        approved_by: Optional[AdminUser] = None,
+    ):
+        if not dt_submitted:
+            dt_submitted = datetime.now()
+        if not dt_updated:
+            dt_updated = datetime.now()
+        new_article = super().create(
+            title=title,
+            content=content,
+            author=author,
+            author_email=author_email,
+            image_url=image_url,
+            summary=summary,
+            dt_submitted=dt_submitted,
+            dt_updated=dt_updated,
+            dt_published=dt_published,
+            state=state,
+            approved_by=approved_by,
+        )
+        new_article.save()
+        return new_article
+
+
+class NewsEntry_Admin(ModelView):
+    model_class = NewsEntry
+
+    def is_accessible(self):
+        return current_user.is_authenticated
+
+
+if not NewsEntry.table_exists():
+    NewsEntry.create_table()
```

Comment:
Comment ça se passe si on change le modèle ?

Reply:
C'est à dire ? genre si on rajoute un truc dans la table ?

Reply:
Si on modifie la table pour rajouter une colonne, il suffit de modifier le modèle dans le fichier et dans une console SQL un `alter table`

Reply:
Après y'a des outils de migration pour peewee mais j'ai du mal avec. cc @debnet ?

D'ailleurs si un de vous a une meilleure idée pour les create table, be my guest :D

Reply:
Ouais, y a peewee-migrations que j'utilise et qui est bien pratique.

https://pypi.org/project/peewee-migrations/

Il s'appuie sur playhouse déjà intégré à peewee, donc c'est smooth et ça offre un utilitaire en ligne de commande `pem` qui supporte même la reversion de migrations à la Django.

Reply:
Top !

Reply:
+1 pour utiliser un système de migrations, car le "tu te ssh et tu alter table a la main" c'est pas très intégrable dans la CI :P

Reply:
Ca marche j'implem ca après, la je fait la modération